components/Helpers.js

```diff
@@ -92,3 +92,14 @@ export function checkEndsWithValidImageExt(imageName) {
   }
   return false;
 }
+
+// checks if url string is valid
+export function isValidHttpUrl(str) {
+  let url;
+  try {
+    url = new URL(str);
+  } catch (_) {
+    return false;  
+  }
+  return url.protocol === "http:" || url.protocol === "https:";

```

Comment:
I'm going to argue for only checking https. 99% of sites today use https and certain browsers don't like non-https sites. We can add an exception later if there is a good reason to, but for now, we should enforce https.